src/Commands/EnvCreateCommand.php

```diff
@@ -150,18 +150,24 @@ public function createBuildEnv(
             $site_name = $site->getName();
             $project = $this->projectFromRemoteUrl($metadata['url']);
             $dashboard_url = "https://dashboard.pantheon.io/sites/{$site_id}#{$multidev}";
+            $pr_id = '';
+            if (preg_match('#^pr-([\d]+)$#', $multidev, $matches)) {
```

Comment:
This is certainly a possibility. Have you considered adding the pull request number as a new parameter to build:env:create? Do we already have a unified PR number available in the CI scripts in example-drops-8-composer & c.? If so, it would be pretty easy to update the CI scripts to pass this option in.

Another potential way to do this would be to ask the repository provider for the PR number, and have each repository provider examine the appropriately-named environment variable with this information. The environment variables set by the CI service should still be available to the Terminus Build Tools plugin commands. This would be quite a bit more code than the existing implementation (or the other proposed alternative), and I am unsure that there would be enough benefit of doing it this way to justify the cost.